R/check_direct_arguments.R

```diff
@@ -164,7 +164,7 @@ direct_check <- function(y,
                      not be included in the indicator. For help, see Example 3
                      in help(direct)."))
       } else if (inherits(custom_indicator[[i]], "function") &&
-        !all(names(formals(custom_indicator[[i]])) ==
+        !all(names(formals(custom_indicator[[i]])) %in%
           c("y", "weights", "threshold"))) {
         stop(strwrap(prefix = " ", initial = "",
                      "Functions for custom indicators need to have exactly the
```

Comment:
Is this still the case? I try to remember but the idea was that you can only give the three arguments but you do not need to? So I can also define a function only with y or am I wrong? 

Maybe as follows (if I understand the new feature correctly):
Functions for custom indicators can only have following three arguments: y, weights and threshold. The target variable y always needs to be specified.

Reply:
else if (inherits(custom_indicator[[i]], "function") &&
        !all(names(formals(custom_indicator[[i]])) %in%
          c("y", "weights", "threshold"))) {
        stop(strwrap(prefix = " ", initial = "",
                     "Functions for custom indicators need to have the argument
                     y and optional the argument weights and threshold. For
                     help, see Example 3 in help(direct)."))
      } else if (inherits(custom_indicator[[i]], "function") &&
        !("y" %in% names(formals(custom_indicator[[i]])))) {
        stop(strwrap(prefix = " ", initial = "",
                     "Functions for custom indicators need to have the argument
                     y and optional the argument weights and threshold. For
                     help, see Example 3 in help(direct)."))
     }